File: app/src/main/java/com/csabacsete/sharedshoppinglist/login/LoginContract.java
package com.csabacsete.sharedshoppinglist.login;

public interface LoginContract {

    interface View {

        String getEmail();

        String getPassword();

        void clearEmailErrors();

        void clearPasswordErrors();

        void showPasswordRequired();

        void showPasswordTooShort();

        void showEmailRequired();

        void showEmailInvalid();

        void showInvalidCredentials();

        void showNetworkError();

        void showCreateAccountError();

        void requestGoogleAccount();

        void showProgress();

        void hideProgress();
    }

    interface Presenter {

        void login();

        void onGooglePlusButtonClicked();

        void onFacebookButtonClicked();

        void onGoogleAccountSignedIn(String idToken);
    }
}

File: app/src/main/java/com/csabacsete/sharedshoppinglist/login/LoginContract.kt
package com.csabacsete.sharedshoppinglist.login

/**
 * Created by ccsete on 10/8/17.
 */
class LoginContract {

    interface View {

        fun getEmail(): String

        fun getPassword(): String

        fun clearEmailErrors()

        fun clearPasswordErrors()

        fun showPasswordRequired()

        fun showPasswordTooShort()

        fun showEmailRequired()

        fun showEmailInvalid()

        fun showInvalidCredentials()

        fun showNetworkError()

        fun requestGoogleAccount()

        fun showProgress()

        fun hideProgress()

        fun goToRegister()
    }

    interface Presenter {

        fun login()

        fun onGooglePlusButtonClicked()

        fun onFacebookButtonClicked()

        fun onGoogleAccountSignedIn(idToken: String)

        fun onRegisterButtonClicked()
    }
}